### ISSUE TITLE
Compile error for coercion to *c_void points to the wrong place when target is a struct field

### ISSUE THREAD

**Issue by SpexGuy** [ziglang/zig#7694]
Tested with 0.7.1
```zig
pub extern fn bar() [*c]u8;

const Struct = struct {
    field: ?*const c_void = null,
};

export fn foo() void {
    const bytes = bar();

    var desc = Struct{};
    desc.field = &bytes;
}
```
```zig
<source>:8:5: error: expected type '*const c_void', found '*const [*c]u8'
    const bytes = stbi_load();
    ^
<source>:8:5: note: pointer type child '[*c]u8' cannot cast into pointer type child 'c_void'
    const bytes = stbi_load();
    ^
<source>:11:19: note: referenced here
    desc.field = &bytes;
                  ^
```

**Comment by LemonBoy** [comment]
```patch
diff --git a/src/stage1/ir.cpp b/src/stage1/ir.cpp
index edda765ff..f274bd51a 100644
--- a/src/stage1/ir.cpp
+++ b/src/stage1/ir.cpp
@@ -20392,7 +20392,7 @@ static IrInstGen *ir_analyze_store_ptr(IrAnalyze *ira, IrInst* source_instr,
     }
 
     ZigType *child_type = ptr->value->type->data.pointer.child_type;
-    IrInstGen *value = ir_implicit_cast(ira, uncasted_value, child_type);
+    IrInstGen *value = ir_implicit_cast2(ira, source_instr, uncasted_value, child_type);
     if (type_is_invalid(value->value->type))
         return ira->codegen->invalid_inst_gen;
 
```

**Comment by mikdusan** [comment]
with latest compiler:
- compile error points to correct location
- appears this was fixed between 0.9.1 and 0.10.0

<details>
<summary>repro.zig</summary>

```zig
pub extern fn bar() [*c]u8;

const Struct = struct {
    field: ?*const anyopaque = null,
};

export fn foo() void {
    const bytes = bar();

    var desc = Struct{};
    desc.field = &bytes;
}
```
</details>

```sh
$ uname -a
Linux archlinux 6.1.2-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 31 Dec 2022 17:40:35 +0000 x86_64 GNU/Linux

$ zig version
0.11.0-dev.1241+0507ced8c

$ zig build-obj repro.zig
repro.zig:12:18: error: expected type '?*const anyopaque', found '*const [*c]u8'
    desc.field = &bytes;
                 ^~~~~~
repro.zig:12:18: note: cannot implicitly cast double pointer '*const [*c]u8' to anyopaque pointer '?*const anyopaque'
```
